src/contract_wrappers/tokens/security_token_wrapper/common.ts

```diff
@@ -2446,8 +2446,25 @@ export default abstract class SecurityTokenCommon extends ERC20TokenWrapper {
     assert.isNonZeroETHAddressHex('Wallet', data.wallet);
     assert.isNonZeroETHAddressHex('ReserveWallet', data.treasuryWallet);
     data.stableTokens.forEach(address => assert.isNonZeroETHAddressHex('stableTokens', address));
-    data.customOracleAddresses.forEach(address => assert.isNonZeroETHAddressHex('customOracleAddresses', address));
-    assert.assert(data.denominatedCurrency !== '', ErrorCode.InvalidData, 'Denominated Currency not provided');
+
+    if (data.customOracleAddresses.length > 0) {
+      if (data.fundRaiseTypes[0] === FundRaiseType.ETH) {
```

Comment:
You shouldn't use `[0]` because you don't know which is the order in `fundRaiseTypes`. You must check if it CONTAINS `FundRaiseType.ETH`